src/state/vaults.ts

```diff
@@ -1,53 +1,107 @@
 import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { range } from 'lodash';
 
-type VaultMetadata = { totalValue: number };
+import { timeUnits } from '@/constants/time';
 
+export interface VaultsState {
+  vaultDetails: VaultDetails;
+  vaultHistory: VaultHistory | undefined;
+  userVault: VaultOwnership | undefined;
+}
 type VaultDetails = {
-  allTimePnl: {
-    percent: number;
-    absolute: number;
-  };
+  totalValue: number;
   thirtyDayReturnPercent: number;
+  positions: Array<VaultPosition>;
+};
+type VaultPosition = {
+  asset: { id: string; name: string };
+  marketId: string;
+  marginUsdc: number;
   currentLeverageMultiple: number;
   currentPosition: {
     asset: number;
     usdc: number;
   };
+  thirtyDayPnl: {
+    percent: number;
+    absolute: number;
+    sparklinePoints: number[];
+  };
 };
-
 type VaultOwnership = {
   userBalance: number;
+  userReturn: {
+    absolute: number;
+    percent: number;
+  };
+  transactionHistory: VaultTransaction[];
+};
+export type VaultTransaction = {
+  timestampMs: number;
+  amountUsdc: number;
+  type: 'withdrawal' | 'deposit';
+  id: string;
+};
+type VaultHistory = {
+  pnlAndEquityDataPoints: Array<{ date: number; equity: number; totalPnl: number }>;
 };
-
-export interface VaultsState {
-  vaults: Record<string, VaultMetadata>;
-  vaultDetails: Record<string, VaultDetails | undefined>;
-  userVaults: Record<string, VaultOwnership | undefined>;
-}
 
 const initialState: VaultsState = {
-  vaults: { 'PEPE-USD': { totalValue: 30_425 } },
   vaultDetails: {
-    'PEPE-USD': {
-      allTimePnl: { absolute: 4_125, percent: 0.1252 },
-      currentLeverageMultiple: 1.2,
-      currentPosition: { asset: 17341235412, usdc: 423.67 },
-      thirtyDayReturnPercent: 0.1474,
-    },
+    totalValue: 30_425,
```

Comment:
[q] is this a default value? or a placeholder for development before BE is complete

Reply:
just placeholder